File: app/src/main/java/de/markhaehnel/rbtv/rocketbeanstv/events/ChannelInfoUpdateEvent.java
package de.markhaehnel.rbtv.rocketbeanstv.events;

import de.markhaehnel.rbtv.rocketbeanstv.objects.RBTV;
import de.markhaehnel.rbtv.rocketbeanstv.objects.schedule.ScheduleItem;
import de.markhaehnel.rbtv.rocketbeanstv.utils.Enums.EventStatus;
import lombok.Getter;

@Getter
public class ChannelInfoUpdateEvent {
    private ScheduleItem scheduleItem;
    private RBTV rbtv;
    private EventStatus status;

    public ChannelInfoUpdateEvent(EventStatus status) {
        this.status = status;
    }

    public ChannelInfoUpdateEvent(ScheduleItem scheduleItem, RBTV rbtv, EventStatus status) {
        this.scheduleItem = scheduleItem;
        this.rbtv = rbtv;
        this.status = status;
    }
}

File: app/src/main/java/de/markhaehnel/rbtv/rocketbeanstv/events/ChannelInfoUpdateEvent.kt
package de.markhaehnel.rbtv.rocketbeanstv.events

import de.markhaehnel.rbtv.rocketbeanstv.objects.RBTV
import de.markhaehnel.rbtv.rocketbeanstv.objects.schedule.ScheduleItem
import de.markhaehnel.rbtv.rocketbeanstv.utils.Enums.EventStatus

class ChannelInfoUpdateEvent {
    var scheduleItem: ScheduleItem = ScheduleItem()
    var rbtv: RBTV = RBTV()
    var status: EventStatus? = null

    constructor(status: EventStatus) {
        this.status = status
    }

    constructor(scheduleItem: ScheduleItem, rbtv: RBTV, status: EventStatus) {
        this.scheduleItem = scheduleItem
        this.rbtv = rbtv
        this.status = status
    }
}
